### PR TITLE
[API] add batch size limit for jsonrpc

### PR DESCRIPTION
**Pull request by millken**
# Description
Limit the count of batch requests in websocket 

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Liuhaai** [comment]
its limited at https://github.com/iotexproject/iotex-core/blob/master/api/websocket.go#L64

**Comment by millken** [comment]
> its limited at https://github.com/iotexproject/iotex-core/blob/master/api/websocket.go#L64

only the message body size is limited here(15M), there needs to limit jsonRPC request array size.

File: api/web3server.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type (
 	}
 
 	web3Handler struct {
+		cfg         Config
```
Comment (dustinxie) [inline api/web3server.go:44]: can just add `batchSizeLimit`, since web3handler does not need the entire `Config`?

File: api/config.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ type Config struct {
 	GasStation      gasstation.Config `yaml:"gasStation"`
 	RangeQueryLimit uint64            `yaml:"rangeQueryLimit"`
 	Tracer          tracer.Config     `yaml:"tracer"`
+	BatchSizeLimit  uint64            `yaml:"batchSizeLimit"`
```
Comment (dustinxie) [inline api/config.go:24]: uint16 is enough for batch size limit

Comment (millken) [reply api/config.go]: done

File: api/web3server.go (hunk 1 above)
Comment (millken) [reply api/web3server.go]: done

**Review by dustinxie — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

File: api/web3server.go [hunk 3]
```diff
@@ -81,10 +82,11 @@ func init() {
 }
 
 // NewWeb3Handler creates a handle to process web3 requests
-func NewWeb3Handler(core CoreService, cacheURL string) Web3Handler {
+func NewWeb3Handler(core CoreService, cfg Config) Web3Handler {
```
Comment (CoderZhi) [inline api/web3server.go:85]: `(core CoreService, limit uint16)`

File: api/web3server.go [hunk 4]
```diff
@@ -41,8 +41,9 @@ type (
 	}
 
 	web3Handler struct {
-		coreService CoreService
-		cache       apiCache
+		batchSizeLimit uint16
```
Comment (CoderZhi) [inline api/web3server.go:44]: `batchSizeLimit int`

File: api/web3server.go [hunk 5]
```diff
@@ -81,10 +82,11 @@ func init() {
 }
 
 // NewWeb3Handler creates a handle to process web3 requests
-func NewWeb3Handler(core CoreService, cacheURL string) Web3Handler {
+func NewWeb3Handler(core CoreService, cfg Config) Web3Handler {
 	return &web3Handler{
-		coreService: core,
-		cache:       newAPICache(15*time.Minute, cacheURL),
+		batchSizeLimit: cfg.BatchSizeLimit,
```
Comment (CoderZhi) [inline api/web3server.go:87]: convert type here

**Review by CoderZhi — APPROVED** [review]

File: api/web3server.go (hunk 3 above)
Comment (millken) [reply api/web3server.go:85]: `cacheURL string ` also from `cfg Config`

**Pull request merged**